pkgs/development/tools/pnpm/fetch-deps/default.nix

```
@@ -36,15 +36,27 @@
         name = "${pname}-pnpm-deps";
 
         nativeBuildInputs = [
+          cacert
           jq
           moreutils
           pnpm
-          cacert
+          yq
         ];
 
         installPhase = ''
           runHook preInstall
 
+          lockfileVersion="$(yq -r .lockfileVersion pnpm-lock.yaml)"
+          if [[ ''${lockfileVersion:0:1} -gt ${lib.versions.major pnpm.version} ]]; then
+            echo "ERROR: lockfileVersion in pnpm-lock.yaml is too new for the provided pnpm version ${lib.versions.major pnpm.version}!"
```

Come to think about it, it's probably also good to print what the lockfileVersion is for convenience
```suggestion
            echo "ERROR: lockfileVersion in pnpm-lock.yaml ($lockfileVersion) is too new for the provided pnpm version (${lib.versions.major pnpm.version})!"
```